File: example/python/random_trader.py
#!/usr/bin/env python3

"""
  Example random trader

  usage random_trader.py --endpoint 'http://localhost:8080' --apikey 'api_key1' --timeout '1000'
"""
import argparse
from fxbattleclient import FxClient, FxClientError
from time import sleep
import random

parser = argparse.ArgumentParser(description='Example random trader')
parser.add_argument('--endpoint',
                    default='http://localhost:8080',
                    help='the api endpoint')
parser.add_argument('--apikey',
                    default='api_key1',
                    help='the api key')
parser.add_argument('--timeout', type=int,
                    default=1000,
                    help='affects the speed of the trader')

args = parser.parse_args()

print("Random Trader", "endpoint:", args.endpoint, "apikey:", args.apikey)
client = FxClient(args.endpoint, args.apikey)
timeout = args.timeout

try:
  while True:
      account = client.account()
      
      if "error" in account:
          print("could not get account details", account["error"])  
          continue
  
      print("account", account)      

      if "USD" in account and account["USD"] > 1:
          account = client.sell("USDGBP", account["USD"])
      elif "GBP" in account and account["GBP"] > 1:
          account = client.sell("GBPUSD", account["GBP"])
      elif "EUR" in account and account["EUR"] > 1:
          account = client.sell("EURGBP", account["EUR"])
      elif "JPY" in account and account["JPY"] > 1:
          account = client.sell("JPYGBP", account["JPY"])
      else:
          print("you're broke!")

      if "error" in account:
          print("failed to sell", account["error"])
      
      sleep(0.1 + ((timeout/1000.0) * random.random()))

except(FxClientError):
    print("error contacting the endpoint")
except(KeyboardInterrupt):
    print()
    print("final holdings", client.account())

File: examples/java/RandomTrader.java
package fxbattle;

import fxbattle.FXBattleClient;

import org.json.JSONObject;
import java.util.concurrent.ThreadLocalRandom;
import com.mashape.unirest.http.exceptions.UnirestException;

public class RandomTrader {
    public static void trade(FXBattleClient client) throws UnirestException, InterruptedException {
        JSONObject acc = client.account();

        if (acc.has("error")) {
            System.out.println(acc.getString("error"));
        }
        else if (acc.has("USD") && acc.getDouble("USD") > 1) {
            acc = client.sell("USDGBP", acc.getDouble("USD"));
        }
        else if (acc.has("GBP") && acc.getDouble("GBP") > 1) {
            acc = client.sell("GBPUSD", acc.getDouble("GBP"));
        }
        else {
            System.out.println("You're broke!");
        }

        if (acc.has("error")) {
            System.out.println("Failed to sell: " + acc.getString("error"));
        }
    }

    public static void main(String[] args) {
        if (args.length != 2) {
            System.out.println("Usage: java -jar trader.jar http://127.0.0.1:8080 api_key");
            return;
        }
        String endpoint = args[0];
        String apiKey = args[1];
        FXBattleClient client = new FXBattleClient(endpoint, apiKey);

        while (true) {
            try {
                trade(client);
                Thread.sleep(ThreadLocalRandom.current().nextInt(100, 1000));
            }
            catch (UnirestException e) {
                System.out.println(e);
            }
            catch (InterruptedException e) {
                System.out.println(e);
            }
        }
    }
}
